File: 1256-rank-transform-of-an-array/rank-transform-of-an-array.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        temp = sorted(arr)
        rank = {}
        r = 1
        for i in temp:
            if i not in rank:
                rank[i] = r
                r += 1

        for i in range(len(arr)):
            arr[i] = rank[arr[i]]
        
        return arr

File: 1256-rank-transform-of-an-array/rank-transform-of-an-array.cpp
class Solution {
public:
    vector<int> arrayRankTransform(vector<int>& arr) {
        vector<int> temp = arr;
        sort(temp.begin(), temp.end());
        unordered_map<int,int> rank;
        int r{1};
        for(int n: temp){
            if(!rank.count(n)) rank[n] = r++;
        }
        for(int &n: arr){
            n = rank[n];
        }
        return arr;
    }
};